### DISCUSSION TITLE
esp.osdebug does nothing

### DISCUSSION THREAD

**Discussion by Tobinator32** [micropython/micropython#12812] [ESP32]
Hello,
according to the documentation it is possible to switch os debugging on by setting esp.osdebug to some of the constants. The Problem is, it is doing nothing. Are there any things i overlook?

**Comment by jimmo** [comment]
@Tobinator32 I think the documentation might be incorrect... looking at the code it should be

```py
esp.osdebug(output, level)
```

where `output=None` disables logging and anything else is supposed to be where the log is sent to. However the output parameter is ignored. It just has to be set to _something_ (e.g. any non-None value, for example `True`).

So to use it, you must do

```py
esp.osdebug(True, esp.LOG_VERBOSE)
```

But even for me that doesn't seem to enable any logging...

**Reply by jimmo** [reply]
Just tested this on v1.19.1, it used to work. Looks like something's broken since the change to IDF 5.0.

Raised #12815

**Reply by Tobinator32** [reply]
where can i download a version that was built with idf4.x ? Do you have a pointer for me? The 1.19 release from the micropython website doesnt seem to work

**Reply by jimmo** [reply]
The working example I posted in https://github.com/micropython/micropython/issues/12815 was using 1.19.1 from the downloads page.

**Comment by shariltumin** [comment]
I've started a discussion #12031 to address the same issue. I've found a heck that makes it possible to see the console log via a custom uart.

I hope this is useful for you, but you will need to build your own firmware.

**Reply by Tobinator32** [reply]
thanks will try it. As this is in an effort to resolve #12819. Have found a solution to get it running but eventially have to come up with a method to get logging info from lwip and wifi. Building own firmware is the least burden to take ;)

**Comment by projectgus** [comment]
Have pushed a PR with a fix: https://github.com/micropython/micropython/pull/12900
